### PR TITLE
Remove dependency with the firstTC and firstDC in the TidbMonitor controller

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
#3630

### What is changed and how does it work?
Remove firsttc and firsetdc dependency, tidbmonitor add timezone field.
InitContainer will not send  `TIDB_ENABLE_BINLOG`,`TIDB_CLUSTER_NAME `,`TIDB_CLUSTER_NAMESPACE ` parameters.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Other side effects: InitContainer will not send  `TIDB_ENABLE_BINLOG`,`TIDB_CLUSTER_NAME `,`TIDB_CLUSTER_NAMESPACE ` parameters.

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Remove dependency with the firstTC and firstDC in the TidbMonitor controller.
```

### CODE REVIEW

**Comment by KanShiori** [comment]
@mikechengwei Please update CRDs because of PR #4151

**Comment by mikechengwei** [comment]
> @mikechengwei Please update CRDs because of PR #4151

done

File: pkg/monitor/monitor/util.go [hunk 1]
```diff
@@ -1215,16 +1180,16 @@ func defaultTidbMonitor(monitor *v1alpha1.TidbMonitor) {
...
 
-func getMonitorStatefulSet(sa *core.ServiceAccount, secret *core.Secret, monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster, dc *v1alpha1.DMCluster, shard int32) (*apps.StatefulSet, error) {
+func getMonitorStatefulSet(sa *core.ServiceAccount, secret *core.Secret, monitor *v1alpha1.TidbMonitor, shard int32) (*apps.StatefulSet, error) {
 	statefulSet := getMonitorStatefulSetSkeleton(sa, monitor, shard)
-	initContainer := getMonitorInitContainer(monitor, tc)
+	initContainer := getMonitorInitContainer(monitor)
 	statefulSet.Spec.Template.Spec.InitContainers = append(statefulSet.Spec.Template.Spec.InitContainers, initContainer)
-	if dc != nil {
-		dmInitContainer := getMonitorDMInitContainer(monitor, dc)
+	if monitor.Spec.DM != nil {
```
Comment (csuzhangxc) [inline pkg/monitor/monitor/util.go:1187]: do we also need to check `monitor.Spec.DM.Clusters > 0`?

Comment (mikechengwei) [reply pkg/monitor/monitor/util.go:1187]: When the number of dm clusters is zero, the `for loop` will not execute . WDYT? @csuzhangxc

File: pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go [hunk 2]
```diff
@@ -180,6 +180,11 @@ type TidbMonitorSpec struct {
 	// if `AlertmanagerURL` is not configured.
 	// +optional
 	EnableAlertRules bool `json:"enableAlertRules,omitempty"`
+
+	// Time zone of TidbMonitor
+	// Optional: Defaults to UTC
+	// +optional
+	Timezone *string `toml:"tz,omitempty" json:"timezone,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go:187]:
```suggestion
	Timezone *string `json:"timezone,omitempty"`
```

Comment (DanielZhangQD) [reply pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go:187]: Is it necessary to use `*string`?

File: pkg/monitor/monitor/util.go [hunk 3]
```diff
@@ -341,26 +345,6 @@ func getMonitorInitContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbClu
...
-			{
-				Name:  "TIDB_ENABLE_BINLOG",
-				Value: strconv.FormatBool(tc.IsTiDBBinlogEnabled()),
-			},
-			{
-				Name:  "TIDB_CLUSTER_NAME",
-				Value: tc.Name,
-			},
-			{
-				Name:  "TIDB_CLUSTER_NAMESPACE",
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/util.go:359]: This cannot be removed now, and is being used in https://github.com/pingcap/monitoring/blob/master/platform-monitoring/operator/init.sh#L116, we need to update the pods.json to support multiple clusters before removing this.

File: pkg/monitor/monitor/util.go [hunk 4]
```diff
@@ -389,17 +373,13 @@ func getMonitorInitContainer(monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbClu
...
 
-func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor, dc *v1alpha1.DMCluster) core.Container {
+func getMonitorDMInitContainer(monitor *v1alpha1.TidbMonitor) core.Container {
 	// TODO: Support dm in reloader. Currently dm cluster shares the same persistent rules dir with tidb cluster
 	command := getInitCommand(monitor)
 	container := core.Container{
 		Name:  "dm-initializer",
 		Image: fmt.Sprintf("%s:%s", monitor.Spec.DM.Initializer.BaseImage, monitor.Spec.DM.Initializer.Version),
 		Env: []core.EnvVar{
-			{
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/util.go:399]: The DM envs are also used in https://github.com/pingcap/ticdc/blob/master/dm/monitoring/init.sh, we have to update that script first.

Comment (DanielZhangQD) [reply pkg/monitor/monitor/util.go:399]: BTW, the `DM_VERSION` is retrieved from the TiDB initializer version instead of DM initializer.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**